### PR TITLE
Introduce `rr.notebook_show()` to simplify notebook experience

### PR DESCRIPTION
**Pull request by jleibs**
### What
As part of adding blueprint support to notebooks, I realized the experience of creating and logging memory recordings felt very incongruous with the normal rerun workflow.

I introduced a new mechanism `rr.notebook_show()`, which uses the MemoryRecording behind the scenes, but doesn't require users to be aware of it.

All Blueprint types now support `_repr_html_` and show the currently active recording stream.

Uncovered several existing sharp-corners with notebooks:
 - MemoryStream was still generating spurious warnings about dropped messages, which I cleaned up.
 - The notebook was using random for its identifiers, which, in the case of examples was having its seed reset, leading to duplicate ids in the DOM.

Lastly, updated the notebook cube example to use the new style:

image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/5715/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/5715/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/5715/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/5715)
- [Docs preview](https://rerun.io/preview/c0177a9fb9cb1ef76f2eeba7b65aa7c861da2642/docs) 
- [Examples preview](https://rerun.io/preview/c0177a9fb9cb1ef76f2eeba7b65aa7c861da2642/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

File: .typos.toml [hunk 1]
```diff
@@ -137,8 +137,9 @@ zeroterminated = "null-terminated"
 zero-terminated = "null-terminated"
 
 [default]
-# Work around for typos inside of 8-character hashes. These show up inside of ipynb.
+extend-ignore-re = ["\"[a-f0-9]{8}\"", "_[a-f0-9]{16}"]
```
Comment (teh-cmc) [inline .typos.toml:140]: Why is the comment after the thing it comments on? :grimacing:

**Review by teh-cmc — APPROVED** [review]
Nice!

- Almost feels like we should add a check to the release checklist which is basically just a readme saying "run the notebook example!".
- The notebook example's README still mention `rr.memory_recording` and such.

---

The last two cells fail with the following for me:
image

even though I'm running everything off of your branch and going through the local webserver:
image

Also I don't understand how I can have a version mismatch if it's all running off of this branch? :thinking: 

---

The embedded viewer is constantly saying `Waiting for data...` even though the data has been loaded:
image

---

Finally, clicking on the Rerun icon in the top left (the main menu) will always crash the view:
image

---

I'm starting to feel like all of these issues are the result of that version mismatch and that I must be doing something wrong but I can't see what?

File: .typos.toml (hunk 1 above)
Comment (jleibs) [reply .typos.toml:140]: Weird, I don't even know how I made that change.

**Comment by teh-cmc** [comment]
> I'm starting to feel like all of these issues are the result of that version mismatch and that I must be doing something wrong but I can't see what?

That was indeed the case.

I had to run `cargo run -p re_build_web_viewer` manually in addition to the usual `just py-build --features web-viewer`, I always forget what is and isn't manual these days... (thanks @jleibs).

This fixes my versioning issues, the last two cells, and even the menu crash! Only the "waiting for data" thingy is left.

**Pull request merged**